modules/c++/cli/unittests/test_cli.cpp

```diff
@@ -175,6 +177,53 @@ TEST_CASE(testRequired)
     TEST_ASSERT_EQ(results->get<std::string>("config"), "configFile");
 }
 
+TEST_CASE(testUnknownArgumentsOptions)
+{
+    cli::ArgumentParser parser(true, std::cerr);
+    parser.setProgram("tester");
+    parser.addArgument("-v --verbose", "Toggle verbose", cli::STORE_TRUE);
+    parser.addArgument("-x --extra", "Extra options", cli::SUB_OPTIONS);
+
+    // Use a flag that is incorrect
+    std::auto_ptr<cli::Results> results(parser.parse(str::split("-z", " ")));
```

Comment:
`std::auto_ptr` is deprecated as of c++11 and removed in c++17, so this should be changed to `std::unique_ptr` to avoid breaking when we update to c++17.